server/src-lib/Hasura/RQL/DDL/Schema.hs

```diff
@@ -56,24 +59,42 @@ import           Hasura.Server.Utils            (matchRegex)
 data RunSQL
   = RunSQL
   { rSql                      :: Text
-  , rCascade                  :: !(Maybe Bool)
+  , rCascade                  :: !Bool
   , rCheckMetadataConsistency :: !(Maybe Bool)
+  , rReadOnly                 :: !Bool
   } deriving (Show, Eq, Lift)
-$(deriveJSON (aesonDrop 1 snakeCase){omitNothingFields=True} ''RunSQL)
+
+instance FromJSON RunSQL where
+  parseJSON (Object o) = do
+    rSql <- o .: "sql"
+    rCascade <- o .:? "cascade" .!= False
+    rCheckMetadataConsistency <- o .:? "check_metadata_consistency"
+    rReadOnly <- o .:? "read_only" .!= False
+    pure RunSQL{..}
+  parseJSON _ = fail "expecting an object for args"
+
+$(deriveToJSON (aesonDrop 1 snakeCase){omitNothingFields=True} ''RunSQL)
 
 runRunSQL :: (CacheBuildM m, UserInfoM m) => RunSQL -> m EncJSON
-runRunSQL (RunSQL t cascade mChkMDCnstcy) = do
+runRunSQL RunSQL{..} = do
   adminOnly
-  isMDChkNeeded <- maybe (isAltrDropReplace t) return mChkMDCnstcy
-  bool (execRawSQL t) (withMetadataCheck (or cascade) $ execRawSQL t) isMDChkNeeded
+  metadataCheckRequired <- case rReadOnly of
+    True -> pure False
+    False -> onNothing rCheckMetadataConsistency $ isAltrDropReplace rSql
```

Comment:
could not set default for `rCheckMetadataConsistency` during parse because if the user explicitly says `false` then doesn't make sense to override that.